modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2375,15 +2376,166 @@ def _dict_func(self, func, axis, *args, **kwargs):
         if "axis" not in kwargs:
             kwargs["axis"] = axis
 
-        def dict_apply_builder(df, func_dict={}):
-            # Sometimes `apply` can return a `Series`, but we require that internally
-            # all objects are `DataFrame`s.
-            return pandas.DataFrame(df.apply(func_dict, *args, **kwargs))
```

Comment:
Why remove the conversion to `pandas.DataFrame`?

Reply:
this is now properly handled by the [`_build_map_reduce`](https://github.com/dchigarev/modin/blob/70a82583576ef02afcf0792c3d641e87bc09e2a3/modin/core/dataframe/pandas/dataframe/dataframe.py#L1639) wrapper that [properly converts](https://github.com/dchigarev/modin/blob/70a82583576ef02afcf0792c3d641e87bc09e2a3/modin/core/dataframe/pandas/dataframe/dataframe.py#L1303-L1317) a Series kernel result into a DataFrame